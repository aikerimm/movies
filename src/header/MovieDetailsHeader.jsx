import './header.css';
import './movieDetails.css';
import MovieDetails from './MovieDetails';
import { useDispatch, useSelector } from 'react-redux';
import { getSelectedMovie, movieSelected } from '../util/moviesSlice.jsx';
import { useCallback } from 'react';
import { useSearchParams } from 'react-router-dom';

const MovieDetailsHeader = () => {
  const dispatch = useDispatch();
  const [searchParams, setSearchParams] = useSearchParams();
  const onSearchClick = useCallback(() => {
    dispatch(movieSelected(null));
    let newSearchParams = new URLSearchParams(searchParams);
    newSearchParams.delete('movie');
    setSearchParams(newSearchParams);
  }, [searchParams, setSearchParams, dispatch]);

  const movie = useSelector(getSelectedMovie);
  return (
    <div className='movieDetailsHeader' id='movieDetailsHeader'>
      <div className='movieDetailsTopPanel'>
        <p className='headerTitle'>
          <b>netflix</b>roulette
        </p>
        <img
          src='/searchIcon.svg'
          alt='searchIcon'
          onClick={onSearchClick}
          className='searchIcon'
          id='searchIcon'
        />
      </div>
      <div className='movieDetailsContent'>
        <MovieDetails movie={movie} />
      </div>
    </div>
  );
};

export default MovieDetailsHeader;
