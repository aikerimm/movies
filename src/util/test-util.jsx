import { configureStore, createAsyncThunk } from '@reduxjs/toolkit';
import moviesReducer from './moviesSlice.jsx';
import { Provider } from 'react-redux';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';

export const testMovie = {
  id: 123,
  title: 'Zootopia',
  vote_average: 9.8,
  runtime: 120,
  genres: ['Drama', 'Comedy'],
  release_date: '1994-01-01',
  overview: 'Lorem Ipsum Some',
  poster_path: '',
};

export const wrapWithReduxAndRouter = (component, initialPath) => {
  const testStore = configureStore({
    reducer: { movies: moviesReducer },
    preloadedState: {
      movies: {
        data: { totalAmount: 1, data: [testMovie], offset: 0, limit: 12 },
        totalAmount: 0,
      },
    },
  });

  const testFetchMovie = createAsyncThunk('movies/fetchMovie', async () => {
    return Promise.resolve(testMovie);
  });

  const testFetchMovies = createAsyncThunk('movies/fetchMovies', async () => {
    return Promise.resolve([testMovie]);
  });

  const testLoader = () => {
    return {
      moviesThunk: testFetchMovies,
      movieThunk: testFetchMovie,
    };
  };

  const testRouter = createMemoryRouter(
    [
      {
        path: '/search',
        element: component,
        loader: testLoader,
      },
    ],
    {
      initialEntries: ['/', initialPath],
      initialIndex: 1,
    }
  );
  const wrappedComponent = (
    <Provider store={testStore}>
      <RouterProvider router={testRouter}></RouterProvider>
    </Provider>
  );
  return [wrappedComponent, testRouter];
};
