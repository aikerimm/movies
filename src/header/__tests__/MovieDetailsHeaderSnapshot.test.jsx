/**
 * @jest-environment jsdom
 */

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MovieDetailsHeader from '../MovieDetailsHeader.jsx';
import moviesReducer from '../../util/moviesSlice.jsx';
import { configureStore } from '@reduxjs/toolkit';
import { testMovie } from '../../util/test-util.jsx';
import { BrowserRouter } from 'react-router-dom';

it('MovieDetailsHeader renders correctly', () => {
  const testStore = configureStore({
    reducer: { movies: moviesReducer },
    preloadedState: {
      movies: {
        selectedMovie: testMovie
      },
    },
  });

  const tree = renderer
    .create(
      <Provider store={testStore}>
        <BrowserRouter>
          <MovieDetailsHeader />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
