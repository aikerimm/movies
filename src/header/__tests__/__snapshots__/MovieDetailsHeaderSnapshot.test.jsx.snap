// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsHeader renders correctly 1`] = `
<div
  className="movieDetailsHeader"
  id="movieDetailsHeader"
>
  <div
    className="movieDetailsTopPanel"
  >
    <p
      className="headerTitle"
    >
      <b>
        netflix
      </b>
      roulette
    </p>
    <img
      alt="searchIcon"
      className="searchIcon"
      id="searchIcon"
      onClick={[Function]}
      src="/searchIcon.svg"
    />
  </div>
  <div
    className="movieDetailsContent"
  >
    <img
      alt="moviePoster"
      className="movieCardPoster"
      src=""
    />
    <div
      className="movieDetails"
    >
      <div
        className="movieDetailsHorizonalDisplay"
      >
        <p
          className="titleText"
          id="movieTitleInHeader"
        >
          Zootopia
        </p>
        <div
          className="circle"
        >
          9.8
        </div>
      </div>
      <p
        className="movieDetailsGenre"
      >
        Drama, Comedy
      </p>
      <div
        className="movieDetailsHorizonalDisplay"
      >
        <p
          className="redText"
        >
          1994
        </p>
        <p
          className="redText"
        >
          2
          h 
          0
          min
        </p>
      </div>
      <p>
        Lorem Ipsum Some
      </p>
    </div>
  </div>
</div>
`;
